parlai/agents/starspace_torch/starspace_torch.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+#
+# Simple implementation of the starspace algorithm, slightly adapted for dialogue.
+# See: https://arxiv.org/abs/1709.03856
+
+from parlai.core.dict import DictionaryAgent
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+from parlai.core.torch_agent import Output
+from .modules import Starspace
+
+import torch
+from torch import optim
+import torch.nn as nn
+import os
+import random
+
+
+class StarspaceTorchAgent(TorchRankerAgent):
+    """Simple implementation of the starspace algorithm: https://arxiv.org/abs/1709.03856
+    """
+
+    OPTIM_OPTS = {
```

Comment:
Do we really need LBFGS and all of these?

TBH I feel like this should really be happening in TorchAgent